R/get_fund_data.R

```diff
@@ -12,35 +14,51 @@
 
 get_fund_data <- function(
   conn,
-  data_timestamp
+  data_timestamp,
+  data_timestamp_lookback
 ) {
   # get the fund holdings and the holdings' reported market value ------------
 
   logger::log_debug("Extracting financial info from database.")
   logger::log_info("using data timestamp: ", data_timestamp)
+  logger::log_debug("Looking back in data to: ", data_timestamp_lookback)
 
   logger::log_trace(
     "Accessing historical fund holdings - security level. ",
     "Filtering to date: ", data_timestamp
   )
   fund_security <- dplyr::tbl(conn, "own_v5_own_fund_detail") %>%
-    dplyr::filter(.data[["report_date"]] == .env[["data_timestamp"]]) %>%
+    dplyr::filter(.data[["report_date"]] <= .env[["data_timestamp"]]) %>%
+    dplyr::filter(
+      .data[["report_date"]] >= .env[["data_timestamp_lookback"]]
+    ) %>%
+    dplyr::group_by(.data$factset_fund_id) %>%
+    dplyr::filter(.data$report_date == max(.data$report_date, na.rm = TRUE)) %>%
+    dplyr::ungroup() %>%
     dplyr::select(
       factset_fund_id = "factset_fund_id",
       holding_fsym_id = "fsym_id",
-      holding_reported_mv = "reported_mv"
+      holding_reported_mv = "reported_mv",
+      report_date = "report_date"
     )
 
   logger::log_trace(
     "Accessing historical fund holdings - non-securities. ",
     "Filtering to date: ", data_timestamp
   )
   fund_nonsecurity <- dplyr::tbl(conn, "own_v5_own_fund_generic") %>%
-    dplyr::filter(.data[["report_date"]] == .env[["data_timestamp"]]) %>%
+    dplyr::filter(.data[["report_date"]] <= .env[["data_timestamp"]]) %>%
+    dplyr::filter(
+      .data[["report_date"]] >= .env[["data_timestamp_lookback"]]
+    ) %>%
```

Comment:
just a question for my own precise understanding... if we have timestamp `2023-12-31` then `data_timestamp - lubridate::dmonths(3L)` = "2023-09-30"? and then if we take any report date that is `>=` to that, we include funds reported on "2023-09-30" (which could maybe be arguably not in Q4)?

Reply:
https://github.com/RMI-PACTA/workflow.factset/issues/57